### ISSUE TITLE
BE players can not connect the server

### ISSUE THREAD

**Issue by dreamLXJ** [GeyserMC/Geyser#4982]
### Describe the bug

In the Bedrock server interface, information such as the server name and the number of players can be displayed. However, upon attempting to join the server, a message indicating the inability to connect to the world is shown.

If Geyser's debug output is enabled, the console will display that someone is pinging and that a connection attempt is being made. However, the attempt simply hangs without any further development or error messages being presented.

### To Reproduce

Try to join the server use BE

### Expected behaviour

Successfully join the server.

### Server Version and Plugins

AnimatedScoreboard, AuraSkills, AuthMe, BetterRTP, BlockLocker, CMILib, CoreProtect, CrossplatForms
, DeluxeMenus, dynmap
*EconomyAPI, Essentials, EssentialsAntiBuild, EssentialsChat, EssentialsProtect, EssentialsSpawn, flo
odgate, Floodgate-Skript, Geyser-Spigot, GeyserHacks
GeyserSkinManager, GraveStonesPlus, GrimAC, JukeBox, KcLoginGui, LiteCommandEditor, LiteSignIn, LuckP
erms, momojs, Multiverse-Core
Multiverse-Inventories, Multiverse-NetherPortals, NoteBlockAPI, PlaceholderAPI, Plan, PlugManX, Prism
, ProtocolLib, Residence, ServerSigns
SkinsRestorer, Skript, Tpa, Vault, ViaBackwards, ViaVersion

### Geyser Dump

https://dump.geysermc.org/K7Uv7IzFWtBSYHdjGHqH6lFk515ckQNT

### Geyser Version

2.4.2-b652 (git-master-b792f72)

### Minecraft: Bedrock Edition Device/Version

1.21.2.02

### Additional Context

The foundational configurations, including the port settings, are correctly established, and the port is operational as expected. Up until approximately one week ago, the Bedrock Edition was capable of successfully accessing the server without encountering any hindrances. Yet, this capability has since ceased. It should be noted that, barring any unforeseen automatic updates, there have been no deliberate plugin updates executed within this timeframe.

**Comment by TheDocu** [comment]
_silence_

**Comment by Kas-tle** [comment]
Can you please use Wireshark to take a packet dump of someone attempting to connect?

**Comment by Kas-tle** [comment]
Could you please provide the pcap file?

**Comment by dreamLXJ** [comment]
> 您能否提供 pcap 文件？

sure,pls
[基岩调试.zip](https://github.com/user-attachments/files/16724771/default.zip)

**Comment by Kas-tle** [comment]
Are you using a vanilla client to make these connections? Based on the capture you have provided, the client is not responding appropriately to the server after it sends Open Connection Reply 1. The correct response by the client here would be Open Connection Request 2. See https://wiki.vg/Raknet_Protocol#Open_Connection_Request_2.

**Comment by dreamLXJ** [comment]
> Are you using a vanilla client to make these connections? Based on the capture you have provided, the client is not responding appropriately to the server after it sends Open Connection Reply 1. The correct response by the client here would be Open Connection Request 2. See https://wiki.vg/Raknet_Protocol#Open_Connection_Request_2.
> 
> Also, have to taken a capture on the client side to confirm that Open Connection Reply 1 is received by the client?


I have indeed noticed this. Judging from the packet capture data from the client I'm sending now, the client has received the response from the server. However, it seems that the packets sent by the client after this are not being received by the server at all. Could this be the reason why I can't connect to the world? Why is this happening, and I'm not the only one who can't connect.

Also, my client is indeed the original version.
[基岩调试（客户端.zip](https://github.com/user-attachments/files/16731081/default.zip)

**Comment by Kas-tle** [comment]
Yeah, based on this it would seem that the client response (Open Connection Request 2) is being dropped somewhere between the client and the server. Does this occur when attempting to join from a client on the same network as the server? Have you tested with clients on networks other than the one in the most recent pcap, and if so is the result the same?

**Comment by dreamLXJ** [comment]
> Yeah, based on this it would seem that the client response (Open Connection Request 2) is being dropped somewhere between the client and the server. Does this occur when attempting to join from a client on the same network as the server? Have you tested with clients on networks other than the one in the most recent pcap, and if so is the result the same?

Yes, I'm unable to connect regardless of which network I use, and it's the same for other players as well.

**Comment by proferabg** [comment]
To whoever can look at this: It seems that the 2 commits to update to [1.21.21](https://github.com/GeyserMC/Geyser/commit/bc7866541008f222554d93cde7ec45c344ed429d#diff-39eb902fb4f2d6bbcbcff5e2d11b08d639bf4ea06c3c25b78cf2f5a164fa3648R53) and the one to [fix ping](https://github.com/GeyserMC/Geyser/commit/b792f72ec70706ce1158d3745fd164e73edf6ae3#diff-5ce20ff2e65b489c04aa2dc58549a9568d1b08d531f95d400305b57b9c0311a5R400), are the ones causing this issue. There is no 1.21.21 bedrock protocol which is confusing the client and server because 1.21.20 is the latest protocol.

**Comment by Camotoy** [comment]
1.21.21 and 1.21.20 share the same protocol. The first commit did have an issue because of the `/` character in the MOTD, however (now fixed). Regardless, the build before those commits are here: https://github.com/GeyserMC/Geyser/actions/runs/10411815722 if you believe these issues did not persist before.

**Comment by proferabg** [comment]
> 1.21.21 and 1.21.20 share the same protocol. The first commit did have an issue because of the `/` character in the MOTD, however (now fixed). Regardless, the build before those commits are here: https://github.com/GeyserMC/Geyser/actions/runs/10411815722 if you believe these issues did not persist before.

I am saying that I have tested B650-b654 and recompiled current commit [2274377](https://github.com/GeyserMC/Geyser/commit/22743778141ffbdf3424842c8f976db2b45d146e) and removed 1.21.21 and only let it send 1.21.20 and it allows joining again without timing out.

**Comment by proferabg** [comment]
For clarification, b650 works, every build after that does not because of sending 1.21.21. Even compiling it to only send 1.21.21 also causes connecting to a server to timeout because the protocol version 712 is 1.21.20 and the bedrock client handles it as 1.21.20 and not 1.21.21.

**Comment by proferabg** [comment]
Actually, after some research, it just seems that because a slash was added to `DEFAULT_BEDROCK_CODEC`, it is freezing up somewhere else in geyser. When hardcoding 1.21.21 or 1.21.20 without the slash, it works with both versions using 1.21.21 bedrock edition client.

**Comment by Camotoy** [comment]
Can you send a Geyser dump from our latest master? The only conclusion I can come to at the moment is some weird JVM or locale behavior.

**Comment by dreamLXJ** [comment]
> Can you send a Geyser dump from our latest master? The only conclusion I can come to at the moment is some weird JVM or locale behavior.

Ok
https://dump.geysermc.org/jg9XisPCoi8Y5OCroiSOOHUMySz1zDol

**Comment by dreamLXJ** [comment]
I haven't encountered this issue after changing the server's hardware, and Bedrock Edition players were able to enter the server normally. However, the problem has resurfaced recently, presenting the same symptoms as before; players can see server population details before attempting to join but are then prompted that they cannot connect to the world upon entry. I've also tried updating to the latest version of Geyser, yet the problem persists. Additionally, the server core remains at version 1.20.4 Paper.

**Comment by Yexeed** [comment]
@dreamLXJ try using waterdog to proxy your player connections to geyser. Solved the problem for me. Waterdog -> Geyser -> Paper.
